types/shares.go

```diff
@@ -59,12 +59,12 @@ func (m Message) MarshalDelimited() ([]byte, error) {
 // appendToShares appends raw data as shares.
 // Used for messages.
 func appendToShares(shares []NamespacedShare, nid namespace.ID, rawData []byte) []NamespacedShare {
-	if len(rawData) < MsgShareSize {
+	if len(rawData) <= MsgShareSize {
```

Comment:
I'm fairly certain now that this needs to be changed to stop splitting up shares of length `MsgShareSize`. I might need to check my sanity though.

Reply:
But why would it need padding in the case `len(rawData) == MsgShareSize `?

Reply:
Can we also have a test that explains that change?

Reply:
I was wrong as this is change actually doesn't alter any of the end results, so it's not needed. If `len(rawData) == MsgShareSize`, then split will only return a single share. I do think the change makes sense from a groking perspective though. I'm happy to change it back, if anyone thinks otherwise.

> But why would it need padding in the case len(rawData) == MsgShareSize ?

it doesn't need padding, but then no padding is added, right? Why push a share through `split` if it's not actually split?

Reply:
Nah, you are right no need to change it back.